python/arcticdb/toolbox/query_stats.py

```diff
@@ -0,0 +1,89 @@
+from contextlib import contextmanager
+import numpy as np
+
+from arcticdb_ext.tools import QueryStats
+from arcticdb_ext.tools.QueryStats import StatsGroupName, StatsName
+
+class QueryStatsTool:
+    # Define enum values as lists since pybind11 enums are not iterable
+    _STATS_NAME_VALUES = [StatsName.result_count, StatsName.total_time_ms, StatsName.count]
+    _STATS_GROUP_NAME_VALUES = [StatsGroupName.arcticdb_call, StatsGroupName.key_type, StatsGroupName.storage_ops]
+
+    @classmethod
+    def context_manager(cls):
+        @contextmanager
+        def _func():
+            cls.enable()
+            yield
+            cls.disable()
+        return _func()
+
+    @classmethod
+    def get_query_stats(cls):
+        # Get raw stats from C++ layer
+        raw_stats = QueryStats.root_layers()
+        
+        # Transform raw stats into structured dictionary
+        result = {}
+        
+        # Process each layer
+        for layer in raw_stats:
+            if layer:
+                cls._process_layer(layer, result)
+            
+        return result
+    
+    @classmethod
+    def _process_layer(cls, layer, current_dict):
+        def _get_enum_name(enum_value):
+            return str(enum_value).split('.')[-1]
+        
+        # Process stats array
+        stats_array = layer.stats
+        for stat_enum in cls._STATS_NAME_VALUES:
+            stat_idx = int(stat_enum)
+            if stats_array[stat_idx] > 0:
+                stat_name = _get_enum_name(stat_enum)
+                if stat_name not in current_dict:
+                    current_dict[stat_name] = stats_array[stat_idx]
+                else:
+                    current_dict[stat_name] += stats_array[stat_idx]
```

Comment:
This is fine while the stats are all counters, just document the limitation carefully